src/services/authorizedItem.service.getItemWithProperties.test.ts

```diff
@@ -0,0 +1,1516 @@
+  afterEach(() => {
+    jest.restoreAllMocks();
+    jest.clearAllMocks();
+  });
+
+  it('Invalid saved membership', async () => {
+    jest.spyOn(itemVisibilityRepository, 'getByItemPath').mockResolvedValue([]);
+
+    jest
+      .spyOn(itemMembershipRepository, 'getInherited')
+      .mockImplementation(
+        async () => ({ permission: 'anything' }) as unknown as ItemMembershipWithItemAndAccount,
+      );
+
+    // any other member shouldn't access
+    await expect(
+      authorizationService.getItemWithProperties(MOCK_DB, {
+        permission: PermissionLevel.Admin,
+        actor: OWNER,
+        item: ITEM,
+      }),
+    ).rejects.toBeInstanceOf(Error);
+  });
+
+  describe('Private item', () => {
+    beforeEach(() => {
+      jest.spyOn(itemVisibilityRepository, 'getByItemPath').mockResolvedValue([]);
+      jest
+        .spyOn(itemMembershipRepository, 'getInherited')
+        .mockImplementation(async (_db, _itemPath, memberId) => {
+          if (memberId === OWNER.id) {
+            return ownerMembership;
+          }
+
+          return null;
+        });
+    });
+
+    it(PermissionLevel.Read, async () => {
+      // owner should pass
+      const { itemMembership: result } = await authorizationService.getItemWithProperties(MOCK_DB, {
+        permission: PermissionLevel.Read,
+        actor: OWNER,
+        item: ITEM,
+      });
+      expect(result).toEqual(ownerMembership);
+
+      // any other member shouldn't access
+      await expect(
+        authorizationService.getItemWithProperties(MOCK_DB, {
+          permission: PermissionLevel.Read,
+          actor: OTHER_MEMBER,
+          item: ITEM,
+        }),
+      ).rejects.toBeInstanceOf(MemberCannotAccess);
+    });
+
+    it(PermissionLevel.Write, async () => {
+      // owner should pass
+      const { itemMembership: result } = await authorizationService.getItemWithProperties(MOCK_DB, {
+        permission: PermissionLevel.Write,
+        actor: OWNER,
+        item: ITEM,
+      });
+      expect(result).toEqual(ownerMembership);
+
+      // any other member shouldn't access
+      await expect(
+        authorizationService.getItemWithProperties(MOCK_DB, {
+          permission: PermissionLevel.Write,
+          actor: OTHER_MEMBER,
+          item: ITEM,
+        }),
+      ).rejects.toBeInstanceOf(MemberCannotAccess);
+    });
+    it(PermissionLevel.Admin, async () => {
+      // owner should pass
+      const { itemMembership: result } = await authorizationService.getItemWithProperties(MOCK_DB, {
+        permission: PermissionLevel.Admin,
+        actor: OWNER,
+        item: ITEM,
+      });
+      expect(result).toEqual(ownerMembership);
+
+      // any other member shouldn't access
+      await expect(
+        authorizationService.getItemWithProperties(MOCK_DB, {
+          permission: PermissionLevel.Admin,
+          actor: OTHER_MEMBER,
+          item: ITEM,
+        }),
+      ).rejects.toBeInstanceOf(MemberCannotAccess);
+    });
```

Comment:
Can these 3 tests be parameterized, so we do not duplicate the logic ? Something like 
`it.each([Permission.Read, Permission.Write, Permission.Admin])(... same code, but use the input value)`